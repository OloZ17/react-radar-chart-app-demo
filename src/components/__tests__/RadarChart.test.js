import React from 'react';
import { render } from '@testing-library/react';
import { RadarChart } from '../index';

describe('Requirements() Component', () => {
  const { ResizeObserver } = window;

  beforeEach(() => {
    delete window.ResizeObserver;
    window.ResizeObserver = jest.fn().mockImplementation(() => ({
      observe: jest.fn(),
      unobserve: jest.fn(),
      disconnect: jest.fn(),
    }));
  });

  afterEach(() => {
    window.ResizeObserver = ResizeObserver;
    jest.restoreAllMocks();
  });

  const data = {
    labels: ['Thing 1', 'Thing 2', 'Thing 3', 'Thing 4', 'Thing 5', 'Thing 6'],
    datasets: [
      {
        label: 'Requirements',
        data: [2, 9, 3, 5, 2, 3],
        backgroundColor: 'rgba(255, 99, 132, 0.2)',
        borderColor: 'rgba(255, 99, 132, 1)',
        borderWidth: 1,
      },
      {
        label: 'Achievements',
        data: [2, 9, 3, 5, 2, 3],
        backgroundColor: 'rgba(12, 8, 255, 0.2)',
        borderColor: 'rgba(12, 8, 255, 1)',
        borderWidth: 1,
      },
    ],
  };

  it('should take a snapshot', () => {
    const { asFragment } = render(<RadarChart data={data} />);
    expect(asFragment(<RadarChart />)).toMatchSnapshot();
  });
});
