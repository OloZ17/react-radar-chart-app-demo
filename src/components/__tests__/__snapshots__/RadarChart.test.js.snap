// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requirements() Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-md-6"
  >
    <canvas
      height="0"
      role="img"
      style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
      width="0"
    />
  </div>
</DocumentFragment>
`;
